jwcrypto/jwa.py

```diff
@@ -485,15 +1492,61 @@ class JWA(object):
         'PS256': _PS256,
         'PS384': _PS384,
         'PS512': _PS512,
-        'none': _None
+        'none': _None,
+        'RSA1_5': _Rsa15,
+        'RSA-OAEP': _RsaOaep,
+        'RSA-OAEP-256': _RsaOaep256,
+        'A128KW': _A128KW,
+        'A192KW': _A192KW,
+        'A256KW': _A256KW,
+        'dir': _Direct,
+        'ECDH-ES': _EcdhEs,
+        'ECDH-ES+A128KW': _EcdhEsAes128Kw,
+        'ECDH-ES+A192KW': _EcdhEsAes192Kw,
+        'ECDH-ES+A256KW': _EcdhEsAes256Kw,
+        'A128GCMKW': _A128GcmKw,
+        'A192GCMKW': _A192GcmKw,
+        'A256GCMKW': _A256GcmKw,
+        'PBES2-HS256+A128KW': _Pbes2Hs256A128Kw,
+        'PBES2-HS384+A192KW': _Pbes2Hs384A192Kw,
+        'PBES2-HS512+A256KW': _Pbes2Hs512A256Kw,
+        'A128CBC-HS256': _A128CbcHs256,
+        'A192CBC-HS384': _A192CbcHs384,
+        'A256CBC-HS512': _A256CbcHs512,
+        'A128GCM': _A128Gcm,
+        'A192GCM': _A192Gcm,
+        'A256GCM': _A256Gcm
     }
 
     @classmethod
     def signing_alg(cls, name):
         try:
             obj = cls.algorithms_registry[name]()
             if obj.algorithm_use != 'sig':
-                raise InvalidJWAAlgorithm(name)
+                raise KeyError()
```

Comment:
Well the idea is that wehn you pass in a use you restrict the virtual set of algorithms used to the subset markled sig, so it's like the other algorithms do not exist for the caller. So I think KeyError is appropriate here (think of it as a virtual sub-dict is being queried). Any error we throw is not seen directly bu the caller anyway as we catch it next and wrap it in a InvalidJWAAlgorithm exception with the proper name and we do not expose the original exception though.